algorithms.py

```diff
@@ -15,58 +17,138 @@
 X = np.random.normal(size = (m, n))
 X = X - np.mean(X, axis=1, keepdims=True)
 
+W1 = np.random.normal(size = (k, m))
+W2 = np.random.normal(size = (m, k))
+
+# auxiliary function that computes the distance between two SVDs
```

Comment:
If you name this `compute_svd_error` then you can delete the comment.
In general, we prefer the code to speak for itself, since comments are not tested or maintainable.